modules/git/git.go

```diff
@@ -206,6 +206,16 @@ func Init(ctx context.Context) error {
 		SupportProcReceive = false
 	}
 
+	if CheckGitVersionAtLeast("2.36") == nil {
+		//
+		// Disable the security check because Gitea runs the git CLI from within the
+		// repository. See https://github.com/go-gitea/gitea/issues/19455 for the full discussion.
+		//
+		if err := checkAndSetConfig("safe.directory", "*", true); err != nil {
+			return err
+		}
+	}
```

Comment:
The version check could be fine tuned. Not only 2.36, many old git versions also have `safe.directory` option.

A more simple approach is that do not do the version check, just write the config for all git versions.

Reply:
I think the `*` option value required git 2.36 thought. The option was introduced in 2.35.x without support for `*`. So I think this check is fine.

Reply:
Nope, there is a discussion

* https://github.com/go-gitea/gitea/pull/19876